corehq/apps/reports/standard/sms.py

```diff
@@ -322,51 +317,121 @@ def uses_locations(self):
 
-            if reporting_locations_id and message.location_id not in reporting_locations_id:
-                continue
+        def get_timestamp(date_):
+            timestamp = ServerTime(date_).user_time(self.timezone).done()
+            table_cell = self._fmt_timestamp(timestamp)
+            return table_cell['html']
 
-            doc_info = self.get_recipient_info(message.couch_recipient_doc_type,
-                message.couch_recipient, contact_cache)
+        def get_contact_link(couch_recipient, couch_recipient_doc_type, raw=False):
+            doc_info = self.get_recipient_info(couch_recipient_doc_type, couch_recipient, contact_cache)
+            table_cell = self._fmt_contact_link(couch_recipient, doc_info)
+            return table_cell['raw'] if raw else table_cell['html']
 
-            phone_number = message.phone_number
-            if abbreviate_phone_number and phone_number is not None:
-                phone_number = phone_number[0:7] if phone_number[0:1] == "+" else phone_number[0:6]
+        startdate = json_format_datetime(self.datespan.startdate_utc)
+        enddate = json_format_datetime(self.datespan.enddate_utc)
+        data = SMS.objects.filter(
+            domain=self.domain,
+            date__range=(startdate, enddate),
+        ).exclude(
+            # Exclude outgoing messages that have not yet been processed
+            direction=OUTGOING,
+            processed=False
+        )
+        data = filter_by_types(data)
+        data = order_by_col(data)
+        if paginate and self.pagination:
+            data = data[self.pagination.start:self.pagination.start + self.pagination.count]
 
-            timestamp = ServerTime(message.date).user_time(self.timezone).done()
-            result.append([
-                self._fmt_timestamp(timestamp),
-                self._fmt_contact_link(message.couch_recipient, doc_info),
-                self._fmt(phone_number),
-                self._fmt_direction(message.direction),
-                self._fmt(message.text),
-                self._fmt(", ".join(message_types)),
-            ])
+        for message in data:
+            yield [
+                get_timestamp(message.date),
+                get_contact_link(message.couch_recipient, message.couch_recipient_doc_type, raw=contact_info),
+                get_phone_number(message.phone_number),
+                get_direction(message.direction),
+                message.text,
+                ', '.join(self._get_message_types(message)),
+            ]
 
-        return result
+    @property
+    def rows(self):
+        return self._get_rows(paginate=True, contact_info=False)
+
+    @property
+    def total_records(self):
+        startdate = json_format_datetime(self.datespan.startdate_utc)
+        enddate = json_format_datetime(self.datespan.enddate_utc)
+        count = SMS.objects.filter(
+            domain=self.domain,
+            date__range=(startdate, enddate),
+        ).count()
```

Comment:
Also, I think the total records needs to have the same filters applied because right now it shows the same count with filters applied and no filters applied.